distributed/utils.py

```diff
@@ -431,6 +431,7 @@ def wait(timeout):
 
     if error is not None:
         typ, exc, tb = error
+        assert exc is not None
```

Comment:
Would this be redundant if you type annotated error on line 393 as

`error: tuple[type[BaseException], BaseException, types.TracebackType] | None = None`
?

Also `sys.exc_info()` is pending deprecation in favour of getting the info from `type(e), e, e.__traceback__`

Reply:
I've found a variant that works if I avoid `sys.exc_info()`.

Reply:
Yours is cleaner, will adjust.

Reply:
Will have to check how the traceback appears in the console log with these different options

Reply:
It looks like a simple `raise error` works as expected.

Reply:
Awesome